nemo/collections/speechlm/models/speech_to_text_llm_model.py

```diff
@@ -0,0 +1,1127 @@
+            (
+                "audio_signal",
+                "audio_signal_length",
+                "processed_signal",
+                "processed_signal_length",
+                "tokens",
+                "tokens_length",
+                "context_start_idx",
+                "num_audios",
+            )
+        )
+    if parallel_state.is_pipeline_last_stage():
+        required_keys.update(("labels", "loss_mask"))
+
+    _batch = {
+        key: move_data_to_device(val, "cuda", non_blocking=True) if key in required_keys and val is not None else None
+        for key, val in _batch.items()
+    }
+
+    # slice batch along sequence dimension for context parallelism
+    output = get_batch_on_this_context_parallel_rank(_batch)
+
+    output["dataloader_idx"] = dataloader_idx
+    output["batch_idx"] = batch_idx
+    return output
+
+
+def speech_to_text_llm_forward_step(model: L.LightningModule, batch: Dict[str, Any]) -> torch.Tensor:
+    forward_args = {
+        "input_ids": batch.get("tokens", None),
+        "input_length": batch.get("tokens_length", None),
+        "loss_mask": batch.get("loss_mask", None),
+        "attention_mask": batch.get("attention_mask", None),
+        "audio_signal": batch.get("audio_signal", None),
+        "audio_signal_length": batch.get("audio_signal_length", None),
+        "processed_signal": batch.get("processed_signal", None),
+        "processed_signal_length": batch.get("processed_signal_length", None),
+        "labels": batch.get("labels", None),
+        "num_audios": batch.get("num_audios", None),
+        "context_start_idx": batch.get("context_start_idx", None),
+        "inference_params": batch.get("inference_params", None),
+    }
+
+    if 'cu_seqlens' in batch:
+        forward_args['packed_seq_params'] = get_packed_seq_params(batch)
+
+    return model(**forward_args)
+
+
+@dataclass
+class SpeechToTextLLMConfig(TransformerConfig, io.IOMixin):
+
+    num_layers: int = 1  # added to avoid init error, not used!!!
+    hidden_size: int = 1  # added to avoid init error, not used!!!
+    num_attention_heads: int = 16  # added to avoid init error, not used!!!
+
+    language_model_config: Optional[GPTConfig] = None
+    language_model_class: Optional[str] = None
+    speech_model_config: Optional[ASRModuleConfig] = None
+    modality_adapter_config: Optional[ModalityAdapterConfig] = None
+
+    language_model_from_pretrained: Optional[str] = None
+    language_model_hub: Optional[str] = 'hf://'
+
+    freeze_language_model: bool = True
+    freeze_speech_model: bool = True
+    freeze_modality_adapter: bool = False
+
+    forward_step_fn: Callable = speech_to_text_llm_forward_step
+    data_step_fn: Callable = speech_to_text_llm_data_step
+
+    text_generation_strategy: SpeechToTextGenerationStrategy = SpeechToTextGenerationStrategy
+
+    inference_config: Optional[Dict[str, Any]] = None
+
+    data_config: Optional[DictConfig] = None
+
+    def freeze_module(self, module: nn.Module):
```

Comment:
duplicate since SpeechLanguageModel has this?

Reply:
You're right, thanks for spotting this~